src/esp/assets/ResourceManager.cpp

```diff
@@ -284,6 +284,40 @@ bool ResourceManager::loadSemanticSceneDescriptor(
   return false;
 }  // ResourceManager::loadSemanticSceneDescriptor
 
+void ResourceManager::buildSemanticColorMap() {
+  CORRADE_ASSERT(semanticScene_,
+                 "Unable to build Semantic Color map due to no semanticScene "
+                 "being loaded.", );
+
+  semanticColorMapBeingUsed_.clear();
+  semanticColorAsInt_.clear();
+  const auto& ssdClrMap = semanticScene_->getSemanticColorMap();
+  if (ssdClrMap.size() == 0) {
+    return;
+  }
+
+  // The color map was built with first maxSemanticID elements in proper order
+  // to match provided semantic IDs (so that ID is IDX of semantic color in
+  // map).  Any overflow colors will be uniquely mapped 1-to-1 to unmapped
+  // semantic IDs as their index.
+  semanticColorMapBeingUsed_.assign(ssdClrMap.begin(), ssdClrMap.end());
+  buildSemanticColorAsIntMap();
+
+}  // ResourceManager::buildSemanticColorMap
+
+void ResourceManager::buildSemanticColorAsIntMap() {
+  semanticColorAsInt_.clear();
+  semanticColorAsInt_.reserve(semanticColorMapBeingUsed_.size());
+  // build listing of colors as ints, with idx being semantic ID
+  std::transform(semanticColorMapBeingUsed_.cbegin(),
+                 semanticColorMapBeingUsed_.cend(),
+                 std::back_inserter(semanticColorAsInt_),
+                 [](const Mn::Color3ub& color) -> uint32_t {
+                   return (unsigned(color[0]) << 16) |
+                          (unsigned(color[1]) << 8) | unsigned(color[2]);
+                 });
```

Comment:
Recurring TODO for me: finally provide a builtin utility to make such nightmare code unnecessary :scream: